### PR TITLE
Med 38 develop registration UI

### PR DESCRIPTION
**Pull request by skazzp**
**Registration page:**

- Added Registration Page UI
- Added Registration form validation
- Added basic Jest tests

[Jira link](https://olhabilous.atlassian.net/browse/MED-38?atlOrigin=eyJpIjoiNDQ2MDRjYThkZWQ5NDBhOGIyYWI2NGZkMDM3MjczZDciLCJwIjoiaiJ9)

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

**Registration page general view:**
image

**Page with validation errors:**
image

### CODE REVIEW

File: apps/medon-fe/src/components/AuthSidebar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Sidebar } from "./styles";
+
+
+export default function AuthSidebar() {
+  return <Sidebar></Sidebar>;
```
Comment (lyudad) [inline apps/medon-fe/src/components/AuthSidebar/index.tsx]: <Sidebar />

File: apps/medon-fe/src/components/AuthSidebar/styles.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import styled from 'styled-components';
+
+export const Sidebar = styled.div`
+  width: 100%;
+  height: auto;
+  min-height: 100vh;
+  background-repeat: no-repeat;
+  background-size: cover;
+  background-image: url('assets/images/sidebar.jpg');
```
Comment (lyudad) [inline apps/medon-fe/src/components/AuthSidebar/styles.tsx]: move to constants

File: apps/medon-fe/src/components/RegistrationForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+      timezone: '(UTC) Coordinated Universal Time',
+    },
+  });
+
+  const role = watch('role');
+
+  const countryOptions = countries.map((country) => {
+    const option = { value: country.name, label: country.name };
+    return option;
+  });
```
Comment (lyudad) [inline apps/medon-fe/src/components/RegistrationForm/index.tsx]: this function will be called with each rerender

File: apps/medon-fe/src/components/RegistrationForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+          <NavLink to="/">
+            <BackBtn type="primary" htmlType="button">
+              {t('regForm.backBtn')}
+            </BackBtn>
+          </NavLink>
+        </BtnContainer>
+      </Form>
+    </Container>
+  );
+}
```
Comment (lyudad) [inline apps/medon-fe/src/components/RegistrationForm/index.tsx:265]: to big

Comment (lyudad) [reply apps/medon-fe/src/components/RegistrationForm/index.tsx:265]: too*

File: apps/medon-fe/src/components/RegistrationForm/styles.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+export const BtnContainer = styled.div`
+  margin-top: 1rem;
+  display: flex;
+  width: 100%;
+  justify-content: space-evenly;
+`;
+
+export const Btn = styled(Button)`
+  background: linear-gradient(90deg, #085dd7 -28.15%, #4d93f8 76.48%);
```
Comment (lyudad) [inline apps/medon-fe/src/components/RegistrationForm/styles.tsx]: all colours in contant file

File: apps/medon-fe/src/pages/RegistrationPage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  Title,
+} from './styles';
+
+export default function RegistrationPage() {
+  const { t } = useTranslation();
+  return (
+    <Container>
+      <RegContainer>
+        <div>
+          <img src={logo} alt="MedOn Logo" />
```
Comment (lyudad) [inline apps/medon-fe/src/pages/RegistrationPage/index.tsx]: all text in text file

Comment (skazzp) [reply apps/medon-fe/src/pages/RegistrationPage/index.tsx]: done

File: apps/medon-fe/src/components/RegistrationForm/styles.tsx (hunk 5 above)
Comment (skazzp) [reply apps/medon-fe/src/components/RegistrationForm/styles.tsx]: moved to theme

File: apps/medon-fe/src/components/RegistrationForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+      role: null,
+      speciality: null,
+      birthday: null,
+      country: null,
+      city: '',
+      timezone: '(UTC) Coordinated Universal Time',
+    },
+  });
+
+  const role = watch('role');
```
[the remark this answers is not in the document]
Comment (skazzp) [reply apps/medon-fe/src/components/RegistrationForm/index.tsx]: changed to constant

File: apps/medon-fe/src/components/AuthSidebar/styles.tsx (hunk 2 above)
Comment (skazzp) [reply apps/medon-fe/src/components/AuthSidebar/styles.tsx]: done

File: apps/medon-fe/src/components/RegistrationForm/index.tsx (hunk 3 above)
Comment (skazzp) [reply apps/medon-fe/src/components/RegistrationForm/index.tsx]: moved to utils

File: apps/medon-fe/src/components/RegistrationForm/index.tsx (hunk 4 above)
Comment (skazzp) [reply apps/medon-fe/src/components/RegistrationForm/index.tsx:265]: split all select inputs to reusable component, moved all options to constants

File: apps/medon-fe/src/validation/registrationFormSchema.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  birthday: yup
+    .date()
+    .nullable()
+    .default(null)
+    .transform((curr, orig) => (orig === '' ? null : curr))
+    .typeError('Birthday is required')
+    .required('Birthday is required'),
+  country: yup.string().required('Country is required'),
+  city: yup.string().required('City is required'),
+  timezone: yup.string().required('Time zone is required'),
```
Comment (lyudad) [inline apps/medon-fe/src/validation/registrationFormSchema.tsx]: can you use here i18n?

Comment (skazzp) [reply apps/medon-fe/src/validation/registrationFormSchema.tsx]: done

**Review by lyudad — APPROVED** [review]

**Pull request merged**